packages/sitecore-jss/src/personalization/personalization-decisions-service.ts

```diff
@@ -0,0 +1,175 @@
+   */
+  currentPageParamsToExtract?: string[];
+  /**
+   * Enables/disables analytics tracking for the Layout Service invocation (default is true).
+   * More than likely, this would be set to false for SSG/hybrid implementations, and the
+   * JSS tracker would instead be used on the client-side: {@link https://jss.sitecore.com/docs/fundamentals/services/tracking}
+   * @default true
+   */
+  tracking?: boolean;
+  /**
+   * The request timeout in milliseconds
+   */
+  timeout?: number;
+  /**
+   * Data fetcher resolver in order to provide custom data fetcher
+   * @see DataFetcherResolver
+   * @see HttpDataFetcher<T>
+   * @see AxiosDataFetcher used by default
+   */
+  dataFetcherResolver?: DataFetcherResolver;
+};
+
+/**
+ * Fetches personalization decisions using the Sitecore REST API.
+ * Uses Axios as the default data fetcher (@see AxiosDataFetcher).
+ */
+export class RestPersonalizationDecisionsService implements PersonalizationDecisionsService {
+  private serviceConfig: RestPersonalizationDecisionsServiceConfig;
+
+  constructor(serviceConfig: RestPersonalizationDecisionsServiceConfig) {
+    this.serviceConfig = {
+      host: '',
+      route: '/sitecore/api/layout/personalization/decision',
+      ...serviceConfig,
+    };
+  }
+
+  /**
+   * Gets personalization decisions.
+   * @param {DecisionsContext} context The decisions context
+   * @returns {Promise<PersonalizationDecisionData>} The personalization decision data
+   */
+  getPersonalizationDecisions(context: DecisionsContext): Promise<PersonalizationDecisionData> {
+    const fetcher = this.serviceConfig.dataFetcherResolver
+      ? this.serviceConfig.dataFetcherResolver<PersonalizationDecisionData>({
+          timeout: this.serviceConfig.timeout,
+        })
+      : this.getDefaultFetcher<PersonalizationDecisionData>();
+    let queryParams = {
+      sc_apikey: this.serviceConfig.apiKey,
+      sc_site: this.serviceConfig.siteName,
+      tracking: this.serviceConfig.tracking ?? true,
+    };
+    if (this.serviceConfig.currentPageParamsToExtract) {
+      queryParams = {
+        ...extractQueryStringParams(
+          window.location.search,
+          this.serviceConfig.currentPageParamsToExtract
+        ),
+        ...queryParams,
+      };
+    }
+    const requestBody = {
+      routePath: context.routePath,
+      language: context.language,
+      renderingIds: context.renderingIds,
+      referrer: window.document.referrer,
+      url: window.location.pathname + window.location.search,
+    };
+
+    debug.personalizationDecisions(
+      'fetching personalization decisions for %o %o',
+      queryParams,
+      requestBody
+    );
+
+    return fetchData<PersonalizationDecisionData>(
+      this.serviceConfig.serviceUrl ?? `${this.serviceConfig.host}${this.serviceConfig.route}`,
+      fetcher,
+      queryParams,
+      requestBody
+    );
+  }
+
+  private getDefaultFetcher = <T>() => {
```

Comment:
Does this need to be generic? See comment above about simplifying data fetcher for this class